Mathlib/Analysis/Convex/Combination.lean

```diff
@@ -590,3 +589,55 @@ Note that `AffineIndependent.convexHull_inter` should be more versatile in most
 lemma AffineIndependent.convexHull_inter' (hs : AffineIndependent R ((↑) : ↑(t₁ ∪ t₂) → E)) :
     convexHull R (t₁ ∩ t₂ : Set E) = convexHull R t₁ ∩ convexHull R t₂ :=
   hs.convexHull_inter subset_union_left subset_union_right
+
+end
+
+section pi
+variable {𝕜 ι : Type*} {E : ι → Type*} [Fintype ι] [LinearOrderedField 𝕜]
+  [Π i, AddCommGroup (E i)] [Π i, Module 𝕜 (E i)] {s : Set ι} {t : Π i, Set (E i)} {x : Π i, E i}
+
+open Finset Fintype
+
+lemma mem_convexHull_pi (h : ∀ i ∈ s, x i ∈ convexHull 𝕜 (t i)) : x ∈ convexHull 𝕜 (s.pi t) := by
+  wlog hs : s = Set.univ
+  · rw [← pi_univ_ite]
+    refine this (fun i _ ↦ ?_) rfl
+    split_ifs with hi
+    · exact h i hi
+    · simp
+  subst hs
+  simp only [Set.mem_univ, mem_convexHull_iff_exists_fintype, true_implies, Set.mem_pi] at h
+  choose κ _ w f hw₀ hw₁ hft hf using h
+  refine mem_convexHull_of_exists_fintype (fun k : Π i, κ i ↦ ∏ i, w i (k i)) (fun g i ↦ f _ (g i))
+    (fun g ↦ prod_nonneg fun _ _ ↦ hw₀ _ _) ?_ (fun _ _ _ ↦ hft _ _) ?_
+  · rw [← Fintype.prod_sum]
+    exact prod_eq_one fun _ _ ↦ hw₁ _
+  ext i
+  calc
+    _ = ∑ g : ∀ i, κ i, (∏ i, w i (g i)) • f i (g i) := by
+      simp only [Finset.sum_apply, Pi.smul_apply]
+    _ = ∑ j : κ i, ∑ g : {g : ∀ k, κ k // g i = j},
+          (∏ k, w k (g.1 k)) • f i ((g : ∀ i, κ i) i) := by
+      rw [← Fintype.sum_fiberwise fun g : ∀ k, κ k ↦ g i]
+    _ = ∑ j : κ i, (∑ g : {g : ∀ k, κ k // g i = j}, ∏ k, w k (g.1 k)) • f i j := by
+      simp_rw [sum_smul]
+      congr! with j _ g _
+      exact g.2
+    _ = ∑ j : κ i, w i j • f i j := ?_
+    _ = x i := hf _
+  congr! with j _
+  calc
+    ∑ g : {g : ∀ k, κ k // g i = j}, ∏ k, w k (g.1 k)
+      = ∑ g ∈ piFinset fun k ↦ if hk : k = i then hk ▸ {j} else univ, ∏ k, w k (g k) :=
+      Finset.sum_bij' (fun g _ ↦ g) (fun g hg ↦ ⟨g, by simpa using mem_piFinset.1 hg i⟩)
+        (by aesop) (by simp) (by simp) (by simp) (by simp)
```

Comment:
Is there a `fiberwise` lemma somewhere for this?

Reply:
Yes I can apply `sum_fiberwise` (or one of its relatives) but there is no way for it to return me the correct RHS (unless I basically write the equality I need as a very specific lemma) so then I need to use `sum_bij` (or one of its relatives) anyway.

If you find a better way, I'm all ears.